src/Spectre.libFunctional/Filter.h

```diff
@@ -33,14 +33,14 @@ namespace spectre::core::functional
 /// <param name="collection">The collection.</param>
 /// <param name="indexes">The indexes.</param>
 /// <returns>Vector consisting of elements under specified indexes in the input collection</returns>
-template <typename DataType>
-std::vector<DataType> filter(gsl::span<const DataType> collection, gsl::span<const size_t> indexes)
+template <typename DataType, typename Index>
+std::vector<DataType> filter(gsl::span<const DataType> collection, gsl::span<const Index> indexes)
```

Comment:
Sudden changes in other projects are always welcome

Reply:
Right, I'll make a PR for that next time :smile:

Reply:
There's not lot of it, so let's take this reasonably :P